diff -ruNp a/common.c b/common.c
--- a/common.c	2021-05-14 21:53:45.745323238 +0900
+++ b/common.c	2021-05-14 21:55:35.879502659 +0900
@@ -323,7 +323,7 @@ void read_sar_cpu_as_paragraph(const cha
         for (i--; (i > 0) && isspace (line[i -1]) ; i--)
             line[i -1] = '\0';
         /* count cpu all line and set numbers to cpu_as_paragraph */
-        if ( strstr ( *(&line), "Average" ) != 0 )
+        if ((strstr (*(&line), "Average") != 0) || (strstr (*(&line), "平均値") != 0)) 
             break;
         if ((strstr (*(&line), "all") != 0)) 
             _cpu_as_paragraph++;
@@ -661,31 +661,31 @@ void sanity_check_file(char * line, cons
     if (strstr(line, "LINUX RESTART") != 0)
         initialize_avg_checked_int ();
 
-    if ((cpu == 1) && (proc_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((cpu == 1) && (proc_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         cpu_avg_checked_int = 1;
         cpu_avg_checked = "Checked";
     }
-    if ((proc_s == 1) && (pswpin_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((proc_s == 1) && (pswpin_s == 0) && ((!strstr(line, "Average")) || (!strstr(line, "平均値")))) {
         proc_s_avg_checked_int = 1;
         proc_s_avg_checked = "Checked";
     }
-    if ((pswpin_s == 1) && (pgpgin_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((pswpin_s == 1) && (pgpgin_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         pswpin_s_avg_checked_int = 1;
         pswpin_s_avg_checked = "Checked";
     }
-    if ((pgpgin_s == 1) && (tps == 0) && (strstr(line, "Average") != 0)) {
+    if ((pgpgin_s == 1) && (tps == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         pgpgin_s_avg_checked_int = 1;
         pgpgin_s_avg_checked = "Checked";
     }
-    if ((tps == 1) && (kbmemfree == 0) && (strstr(line, "Average") != 0)) {
+    if ((tps == 1) && (kbmemfree == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         tps_avg_checked_int = 1;
         tps_avg_checked = "Checked";
     }
-    if ((kbmemfree == 1) && (kbswpfree == 0) && (strstr(line, "Average") != 0)) {
+    if ((kbmemfree == 1) && (kbswpfree == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         kbmemfree_avg_checked_int = 1;
         kbmemfree_avg_checked = "Checked";
     }
-    if ((kbswpfree == 1) && (strstr(line, "Average") != 0)) {
+    if ((kbswpfree == 1) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         if ((kbhugfree == 1) || (dentunusd == 0)) {
             kbswpfree_avg_checked_int = 1;
             kbswpfree_avg_checked = "Checked";
@@ -693,39 +693,39 @@ void sanity_check_file(char * line, cons
             kbhugfree_avg_checked = "Checked";
         }
     }
-    if ((kbhugfree == 1) && (dentunusd == 0) && (strstr(line, "Average") != 0)) {
+    if ((kbhugfree == 1) && (dentunusd == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         kbhugfree_avg_checked_int = 1;
         kbhugfree_avg_checked = "Checked";
     }
-    if ((dentunusd == 1) && (runq_sz == 0) && (strstr(line, "Average") != 0)) {
+    if ((dentunusd == 1) && (runq_sz == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         dentunusd_avg_checked_int = 1;
         dentunusd_avg_checked = "Checked";
     }
-    if ((runq_sz == 1) && (dev == 0) && (strstr(line, "Average") != 0)) {
+    if ((runq_sz == 1) && (dev == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         runq_sz_avg_checked_int = 1;
         runq_sz_avg_checked = "Checked";
     }
-    if ((dev == 1) && (rxpck_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((dev == 1) && (rxpck_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         dev_avg_checked_int = 1;
         dev_avg_checked = "Checked";
     }
-    if ((rxpck_s == 1) && (rxerr_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((rxpck_s == 1) && (rxerr_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         rxpck_s_avg_checked_int = 1;
         rxpck_s_avg_checked = "Checked";
     }
-    if ((rxerr_s == 1) && (call_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((rxerr_s == 1) && (call_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         rxerr_s_avg_checked_int = 1;
         rxerr_s_avg_checked = "Checked";
     }
-    if ((call_s == 1) && (scall_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((call_s == 1) && (scall_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         call_s_avg_checked_int = 1;
         call_s_avg_checked = "Checked";
     }
-    if ((scall_s == 1) && (totsck == 0) && (strstr(line, "Average") != 0)) {
+    if ((scall_s == 1) && (totsck == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         scall_s_avg_checked_int = 1;
         scall_s_avg_checked = "Checked";
     }
-    if ((totsck == 1) && (strstr(line, "Average") != 0)) {
+    if ((totsck == 1) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         if (squeezd == 0) {
             totsck_avg_checked_int = 1;
             totsck_avg_checked = "Checked";
@@ -733,7 +733,7 @@ void sanity_check_file(char * line, cons
             squeezd_avg_checked = "Checked";
         }
     }
-    if ((squeezd == 1) && (strstr(line, "Average") != 0)) {
+    if ((squeezd == 1) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         squeezd_avg_checked_int = 1;
         squeezd_avg_checked = "Checked";
     }
diff -ruNp a/sar_analyzer.c b/sar_analyzer.c
--- a/sar_analyzer.c	2021-05-14 21:47:10.184808276 +0900
+++ b/sar_analyzer.c	2021-05-14 21:56:57.111938958 +0900
@@ -154,7 +154,7 @@ int set_token_column(int file_number, ch
         }
         /* if AM or PM will appear, column number should be decremented, unless, segfault in set_column_item() */
         if (SAR_OPTION != 'Z') {
-            if ((strstr(token, "AM") != NULL) || (strstr(token, "PM") != NULL))
+            if ((strstr(token, "AM") != NULL) ||(strstr(token, "PM") != NULL) ||(strstr(token, "午前") != NULL) ||(strstr(token, "午後") != NULL)) 
                 i--;
         }
         /* this item name is set by get_word_line() */
@@ -582,7 +582,7 @@ int set_token_items(int file_number, cha
 
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -645,12 +645,12 @@ int set_token_items(int file_number, cha
                         set_cpu_lowest_time(time_value, utility, "usr");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -692,7 +692,7 @@ int set_token_items(int file_number, cha
             } else if ((i == get_column_cpu("sys")) &&(strstr(token, "%sys") == 0)) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -755,12 +755,12 @@ int set_token_items(int file_number, cha
                         set_cpu_lowest_time(time_value, utility, "sys");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -802,7 +802,7 @@ int set_token_items(int file_number, cha
             } else if ((i == get_column_cpu("iowait")) &&(strstr(token, "%iowait") == 0)) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -865,12 +865,12 @@ int set_token_items(int file_number, cha
                         set_cpu_lowest_time(time_value, utility, "iowait");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -912,7 +912,7 @@ int set_token_items(int file_number, cha
             } else if ((i == get_column_cpu("idle")) &&(strstr(token, "%idle") == 0)) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -975,12 +975,12 @@ int set_token_items(int file_number, cha
                         set_cpu_lowest_time(time_value, utility, "idle");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -1050,7 +1050,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_tasks("proc")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1130,7 +1130,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_tasks("cswch")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1238,7 +1238,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_pswap("pswpin")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1318,7 +1318,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_pswap("pswpout")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1462,7 +1462,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_paging("pgpgin")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1543,7 +1543,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_paging("pgpgout")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1624,7 +1624,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_paging("fault")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1704,7 +1704,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_paging("majflt")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1784,7 +1784,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_paging("vmeff")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1905,7 +1905,7 @@ int set_token_items(int file_number, cha
            if (i == get_column_io_transfer_rate("tps")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1985,7 +1985,7 @@ int set_token_items(int file_number, cha
            } else if (i == get_column_io_transfer_rate( "bread")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2065,7 +2065,7 @@ int set_token_items(int file_number, cha
            } else if (i == get_column_io_transfer_rate("bwrtn")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2185,7 +2185,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_memory("memused")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2249,12 +2249,12 @@ int set_token_items(int file_number, cha
                         set_memory_lowest_time(time_value, "memused");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -2294,7 +2294,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_memory("kbcommit")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2374,7 +2374,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_memory("commit")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2470,7 +2470,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_swpused()) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2533,12 +2533,12 @@ int set_token_items(int file_number, cha
                         set_swpused_lowest_time(time_value);
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -2618,7 +2618,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_kernel_table("dentunusd")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2698,7 +2698,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_kernel_table("file")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2778,7 +2778,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_kernel_table("inode")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2924,7 +2924,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_ldavg("runq_sz")) {
                 if (SAR_OPTION == 'Z') {
                     t = atoi(token);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3005,7 +3005,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_ldavg("plist_sz")) {
                 if (SAR_OPTION == 'Z') {
                     t = atoi(token);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3086,7 +3086,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_ldavg("ldavg_one")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3149,12 +3149,12 @@ int set_token_items(int file_number, cha
                         set_ldavg_lowest_time(time_value, "ldavg_one");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -3194,7 +3194,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_ldavg("ldavg_five")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3257,12 +3257,12 @@ int set_token_items(int file_number, cha
                         set_ldavg_lowest_time( time_value, "ldavg_five");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -3302,7 +3302,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_ldavg("ldavg_15")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3365,12 +3365,12 @@ int set_token_items(int file_number, cha
                         set_ldavg_lowest_time(time_value, "ldavg_15");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -3430,7 +3430,7 @@ int set_token_items(int file_number, cha
            if (i == get_column_block_device("areqsz")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3491,7 +3491,7 @@ int set_token_items(int file_number, cha
            } else if (i == get_column_block_device("util")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3536,12 +3536,12 @@ int set_token_items(int file_number, cha
                         set_block_device_lowest_time(time_value, utility, "util");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_null[MAX_LINE_LENGTH] = {'\0'};
@@ -3622,7 +3622,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_network("rxpck")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3688,7 +3688,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("txpck")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3754,7 +3754,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("rxkb")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3820,7 +3820,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("txkb")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3922,7 +3922,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_network("rxerr")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3982,7 +3982,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("txerr")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -4043,7 +4043,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("rxdrop")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -4104,7 +4104,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("txdrop")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -4173,7 +4173,7 @@ int get_word_line(int file_number, char
     int PM = 0;
     /* for analyzing all data */
     if (SAR_OPTION == 'Z') {
-        if (strstr(*line, "PM") != NULL)
+        if ((strstr(*line, "PM") != NULL) ||(strstr(*line, "午後") != NULL)) 
             PM = 1;
         /* testing */
         char *str_tmp = "";
@@ -4194,7 +4194,7 @@ int get_word_line(int file_number, char
              */
             initialize_check_int();
         /* first, we are setting column number of interests */
-        } else if ((strstr(*line, "Average")) == 0 && (strstr(*line, "usr") || strstr(*line, "sys")   != NULL)) {
+        } else if (((strstr(*line, "Average") == 0) ||(strstr(*line, "平均値") == 0))  &&((strstr(*line, "usr") || strstr(*line, "sys") != NULL))) {
             if (MESSAGE_ONLY == 0)
                 append_list(&line_all_obj, str_tmp);
 
@@ -4267,7 +4267,7 @@ int get_word_line(int file_number, char
             PM = set_token_column(file_number, *line, "runq", 0, SAR_OPTION);
             /* this means, skip this clause on next loop, it's enough to set set_token_column() one time */
             CHK_Z = _LDAVG;
-        } else if ((strstr(*line, "Average")) == 0 && strstr(*line,"DEV") != NULL) {
+        } else if (((strstr(*line, "Average") == 0) ||(strstr(*line, "平均値") == 0))  &&(strstr(*line, "DEV") != NULL)) {
             if (MESSAGE_ONLY == 0)
                 append_list(&line_all_obj, str_tmp);
 
@@ -4275,7 +4275,7 @@ int get_word_line(int file_number, char
             PM = set_token_column(file_number, *line, "DEV", 0, SAR_OPTION);
             /* this means, skip this clause on next loop, it's enough to set set_token_column() one time */
             CHK_Z = _BLOCK_DEVICE;
-        } else if ((strstr(*line, "Average")) == 0 && strstr(*line,"rxpck/s") != NULL) {
+        } else if (((strstr(*line, "Average") == 0) ||(strstr(*line, "平均値") == 0)) &&(strstr(*line,"rxpck/s") != NULL)) {
             if (MESSAGE_ONLY == 0)
                 append_list(&line_all_obj, str_tmp);
 
@@ -4283,7 +4283,7 @@ int get_word_line(int file_number, char
             PM = set_token_column(file_number, *line, "rxpck", 0, SAR_OPTION);
             /* this means, skip this clause on next loop, it's enough to set set_token_column() one time */
             CHK_Z = _NETWORK;
-        } else if ((strstr(*line, "Average")) == 0 && strstr (*line,"rxerr/s") != NULL) {
+        } else if (((strstr(*line, "Average") == 0) ||(strstr(*line, "平均値") == 0)) &&(strstr(*line,"rxerr/s") != NULL)) {
             if (MESSAGE_ONLY == 0)
                 append_list(&line_all_obj, str_tmp);
 
@@ -4319,7 +4319,7 @@ int get_word_line(int file_number, char
                     CHK_CORES_N = 0;
                 }
                 CHECK_CPU_EACH++;
-            } else if (strstr(*line, "Average") != 0) {
+            } else if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) {
                 CHK_Z &= ~(1 << 0);  /* this means, skip this clause on next loop */
                 CHK_CORES_N = 0;
                 /* set_token_items(file_number, line, "cpu" , CHK_CORES_N, SAR_OPTION); */
@@ -4337,7 +4337,7 @@ int get_word_line(int file_number, char
             }
         /* setting utility arguments to meaningless value like -999 */
         } else if (CHK_Z == _TASKS) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 1);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4346,7 +4346,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "proc", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _PSWAP) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 2);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4355,7 +4355,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "pswpin/s", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _PAGING) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 3);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4364,7 +4364,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "fault", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _IO_TRANSFER_RATE) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 4);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4373,7 +4373,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "bread/s", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _MEMORY) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 5);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4382,7 +4382,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "kbmemfree", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _SWPUSED) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 6);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4391,7 +4391,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "kbswpfree", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _KERNEL_TABLE) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 7);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4400,7 +4400,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "dentunusd", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _LDAVG) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 8);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4413,7 +4413,7 @@ int get_word_line(int file_number, char
          * setting each name to the object for future use 
          */ 
         } else if (CHK_Z == _BLOCK_DEVICE) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 9);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4426,7 +4426,7 @@ int get_word_line(int file_number, char
          * setting each name to the object for future use 
          */ 
         } else if (CHK_Z == _NETWORK) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 10);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4435,7 +4435,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "rxpck" , -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _NETWORK_ERROR) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 11);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4458,7 +4458,8 @@ int get_word_line(int file_number, char
             && strstr(*line, "rcvin/s") == 0 && strstr(*line, "cswch/s") == 0 
             && strstr(*line, "frmpg/s") == 0 && strstr(*line, "call/s") == 0 
             && strstr(*line, "scall/s") == 0 && strstr(*line, "totsck") == 0 
-            && strstr(*line, "total/s") == 0 && strstr(*line, "Average") == 0) {
+            && strstr(*line, "total/s") == 0 && strstr(*line, "Average") == 0
+            && strstr(*line, "平均") == 0) {
             /* no match */
             return 0;
         } else {
@@ -4758,7 +4759,7 @@ int get_word_line(int file_number, char
              * cpu and sets core number every time. core number is incremented, so, lately, core number could be known (including all line)  
              * stop calling this clause if reached up to MAX_CORE_NUMBERS, anyway, it will no be called very soon   
              */
-            if (strstr(*line, "Average") && !strstr(*line, "CPU") && (SHOW_AVG == 2) && (CHK_CORES_N <= MAX_CORE_NUMBERS)) {
+            if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&  !strstr(*line, "CPU") &&(SHOW_AVG == 2) &&(CHK_CORES_N <= MAX_CORE_NUMBERS)) {
                 append_list(&line_obj, str_tmp);
                 set_token_items(file_number, line, "cpu" , CHK_CORES_N, SAR_OPTION, time_span, PM); /* setting each cpu values to the object */
                 if (CHK_CORES_N == MAX_CORE_NUMBERS)
@@ -4769,7 +4770,7 @@ int get_word_line(int file_number, char
              * just because not knowing which block device will appear every time this clause is called,
              * setting each name to the object for future use 
              */ 
-            } else if (strstr(*line, "Average")  && !strstr(*line, "DEV") && (SHOW_AVG == 3)) {
+            } else if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&  !strstr(*line, "DEV") &&(SHOW_AVG == 3)) {
                 if (strstr(*line, "dev") != 0) {
                     append_list(&line_obj, str_tmp);
                     set_token_items(file_number, line, "DEV" , -999, SAR_OPTION, time_span, PM);
@@ -4778,17 +4779,17 @@ int get_word_line(int file_number, char
              * just because not knowing which network device will appear every time this clause is called,
              * setting each name to the object for future use 
              */ 
-            } else if (strstr(*line, "Average")  && !strstr(*line, "rxpck") && (SHOW_AVG == 4)) {
+            } else if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&  !strstr(*line, "rxpck") &&(SHOW_AVG == 4)) {
                 append_list(&line_obj, str_tmp);
                 set_token_items(file_number, line, "rxpck" , -999, SAR_OPTION, time_span, PM);
-            } else if (strstr(*line, "Average")  && !strstr(*line, "rxerr") && (SHOW_AVG == 5)) {
+            } else if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&  !strstr(*line, "rxerr") &&(SHOW_AVG == 5)) {
                 append_list(&line_obj, str_tmp);
                 set_token_items(file_number, line, "rxerr" , -999, SAR_OPTION, time_span, PM);
 
             /* Call set_token_items() here
              * this clause is for others which only have one line to be analyzed 
              */
-            } else if (strstr(*line, "Average")  && ((SHOW_AVG == 0)) && (strstr(*line, "CPU") == NULL)) {
+            } else if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&((SHOW_AVG == 0)) &&(strstr(*line, "CPU") == NULL)) {
                 append_list(&line_obj, str_tmp);
                 /* setting utility arguments to meaningless value line -999 */
                 if (((CHK_A >> 1) & 1)) {
