diff -ruNp a/common.c b/common.c
--- a/common.c	2021-05-05 17:10:15.207794405 +0900
+++ b/common.c	2021-05-05 17:10:31.804467552 +0900
@@ -298,7 +298,7 @@ void read_sar_cpu_as_paragraph(const cha
         for (i--; (i > 0) && isspace (line[i -1]) ; i--)
             line[i -1] = '\0';
         /* count cpu all line and set numbers to cpu_as_paragraph */
-        if ( strstr ( *(&line), "Average" ) != 0 )
+        if ((strstr (*(&line), "Average") != 0) || (strstr (*(&line), "平均値") != 0)) 
             break;
         if ((strstr (*(&line), "all") != 0)) 
             _cpu_as_paragraph++;
@@ -627,31 +627,31 @@ void sanity_check_file(char * line, cons
     if (strstr(line, "LINUX RESTART") != 0)
         initialize_avg_checked_int ();
 
-    if ((cpu == 1) && (proc_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((cpu == 1) && (proc_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         cpu_avg_checked_int = 1;
         cpu_avg_checked = "Checked";
     }
-    if ((proc_s == 1) && (pswpin_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((proc_s == 1) && (pswpin_s == 0) && ((!strstr(line, "Average")) || (!strstr(line, "平均値")))) {
         proc_s_avg_checked_int = 1;
         proc_s_avg_checked = "Checked";
     }
-    if ((pswpin_s == 1) && (pgpgin_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((pswpin_s == 1) && (pgpgin_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         pswpin_s_avg_checked_int = 1;
         pswpin_s_avg_checked = "Checked";
     }
-    if ((pgpgin_s == 1) && (tps == 0) && (strstr(line, "Average") != 0)) {
+    if ((pgpgin_s == 1) && (tps == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         pgpgin_s_avg_checked_int = 1;
         pgpgin_s_avg_checked = "Checked";
     }
-    if ((tps == 1) && (kbmemfree == 0) && (strstr(line, "Average") != 0)) {
+    if ((tps == 1) && (kbmemfree == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         tps_avg_checked_int = 1;
         tps_avg_checked = "Checked";
     }
-    if ((kbmemfree == 1) && (kbswpfree == 0) && (strstr(line, "Average") != 0)) {
+    if ((kbmemfree == 1) && (kbswpfree == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         kbmemfree_avg_checked_int = 1;
         kbmemfree_avg_checked = "Checked";
     }
-    if ((kbswpfree == 1) && (strstr(line, "Average") != 0)) {
+    if ((kbswpfree == 1) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         if ((kbhugfree == 1) || (dentunusd == 0)) {
             kbswpfree_avg_checked_int = 1;
             kbswpfree_avg_checked = "Checked";
@@ -659,39 +659,39 @@ void sanity_check_file(char * line, cons
             kbhugfree_avg_checked = "Checked";
         }
     }
-    if ((kbhugfree == 1) && (dentunusd == 0) && (strstr(line, "Average") != 0)) {
+    if ((kbhugfree == 1) && (dentunusd == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         kbhugfree_avg_checked_int = 1;
         kbhugfree_avg_checked = "Checked";
     }
-    if ((dentunusd == 1) && (runq_sz == 0) && (strstr(line, "Average") != 0)) {
+    if ((dentunusd == 1) && (runq_sz == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         dentunusd_avg_checked_int = 1;
         dentunusd_avg_checked = "Checked";
     }
-    if ((runq_sz == 1) && (dev == 0) && (strstr(line, "Average") != 0)) {
+    if ((runq_sz == 1) && (dev == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         runq_sz_avg_checked_int = 1;
         runq_sz_avg_checked = "Checked";
     }
-    if ((dev == 1) && (rxpck_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((dev == 1) && (rxpck_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         dev_avg_checked_int = 1;
         dev_avg_checked = "Checked";
     }
-    if ((rxpck_s == 1) && (rxerr_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((rxpck_s == 1) && (rxerr_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         rxpck_s_avg_checked_int = 1;
         rxpck_s_avg_checked = "Checked";
     }
-    if ((rxerr_s == 1) && (call_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((rxerr_s == 1) && (call_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         rxerr_s_avg_checked_int = 1;
         rxerr_s_avg_checked = "Checked";
     }
-    if ((call_s == 1) && (scall_s == 0) && (strstr(line, "Average") != 0)) {
+    if ((call_s == 1) && (scall_s == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         call_s_avg_checked_int = 1;
         call_s_avg_checked = "Checked";
     }
-    if ((scall_s == 1) && (totsck == 0) && (strstr(line, "Average") != 0)) {
+    if ((scall_s == 1) && (totsck == 0) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         scall_s_avg_checked_int = 1;
         scall_s_avg_checked = "Checked";
     }
-    if ((totsck == 1) && (strstr(line, "Average") != 0)) {
+    if ((totsck == 1) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         if (squeezd == 0) {
             totsck_avg_checked_int = 1;
             totsck_avg_checked = "Checked";
@@ -699,7 +699,7 @@ void sanity_check_file(char * line, cons
             squeezd_avg_checked = "Checked";
         }
     }
-    if ((squeezd == 1) && (strstr(line, "Average") != 0)) {
+    if ((squeezd == 1) && ((strstr(line, "Average") != 0) || (strstr(line, "平均値") != 0))) {
         squeezd_avg_checked_int = 1;
         squeezd_avg_checked = "Checked";
     }
diff -ruNp a/sar_analyzer.c b/sar_analyzer.c
--- a/sar_analyzer.c	2021-05-05 17:10:22.445651863 +0900
+++ b/sar_analyzer.c	2021-05-05 17:10:45.017207339 +0900
@@ -141,7 +141,7 @@ int set_token_column(int file_number, ch
         }
         /* if AM or PM will appear, column number should be decremented, unless, segfault in set_column_item() */
         if (SAR_OPTION != 'Z') {
-            if ((strstr(token, "AM") != NULL) || (strstr(token, "PM") != NULL))
+            if ((strstr(token, "AM") != NULL) ||(strstr(token, "PM") != NULL) ||(strstr(token, "午前") != NULL) ||(strstr(token, "午後") != NULL)) 
                 i--;
         }
         /* this item name is set by get_word_line() */
@@ -553,7 +553,7 @@ int set_token_items(int file_number, cha
 
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -616,12 +616,12 @@ int set_token_items(int file_number, cha
                         set_cpu_lowest_time(time_value, utility, "usr");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_echo3[MAX_LINE_LENGTH] = {'\0'};
@@ -652,7 +652,7 @@ int set_token_items(int file_number, cha
             } else if ((i == get_column_cpu("sys")) &&(strstr(token, "%sys") == 0)) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -715,12 +715,12 @@ int set_token_items(int file_number, cha
                         set_cpu_lowest_time(time_value, utility, "sys");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_echo3[MAX_LINE_LENGTH] = {'\0'};
@@ -751,7 +751,7 @@ int set_token_items(int file_number, cha
             } else if ((i == get_column_cpu("iowait")) &&(strstr(token, "%iowait") == 0)) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -814,12 +814,12 @@ int set_token_items(int file_number, cha
                         set_cpu_lowest_time(time_value, utility, "iowait");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_echo3[MAX_LINE_LENGTH] = {'\0'};
@@ -850,7 +850,7 @@ int set_token_items(int file_number, cha
             } else if ((i == get_column_cpu("idle")) &&(strstr(token, "%idle") == 0)) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -913,12 +913,12 @@ int set_token_items(int file_number, cha
                         set_cpu_lowest_time(time_value, utility, "idle");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_echo3[MAX_LINE_LENGTH] = {'\0'};
@@ -977,7 +977,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_tasks("proc")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1057,7 +1057,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_tasks("cswch")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1165,7 +1165,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_pswap("pswpin")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1245,7 +1245,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_pswap("pswpout")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1389,7 +1389,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_paging("pgpgin")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1470,7 +1470,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_paging("pgpgout")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1551,7 +1551,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_paging("fault")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1631,7 +1631,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_paging("majflt")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1711,7 +1711,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_paging("vmeff")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1832,7 +1832,7 @@ int set_token_items(int file_number, cha
            if (i == get_column_io_transfer_rate("tps")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1912,7 +1912,7 @@ int set_token_items(int file_number, cha
            } else if (i == get_column_io_transfer_rate( "bread")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -1992,7 +1992,7 @@ int set_token_items(int file_number, cha
            } else if (i == get_column_io_transfer_rate("bwrtn")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2112,7 +2112,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_memory("memused")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2176,12 +2176,12 @@ int set_token_items(int file_number, cha
                         set_memory_lowest_time(time_value, "memused");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_echo3[MAX_LINE_LENGTH] = {'\0'};
@@ -2210,7 +2210,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_memory("kbcommit")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2290,7 +2290,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_memory("commit")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2386,7 +2386,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_swpused()) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2449,12 +2449,12 @@ int set_token_items(int file_number, cha
                         set_swpused_lowest_time(time_value);
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_echo3[MAX_LINE_LENGTH] = {'\0'};
@@ -2522,7 +2522,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_kernel_table("dentunusd")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2602,7 +2602,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_kernel_table("file")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2682,7 +2682,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_kernel_table("inode")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2828,7 +2828,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_ldavg("runq_sz")) {
                 if (SAR_OPTION == 'Z') {
                     t = atoi(token);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2909,7 +2909,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_ldavg("plist_sz")) {
                 if (SAR_OPTION == 'Z') {
                     t = atoi(token);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -2990,7 +2990,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_ldavg("ldavg_one")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3070,7 +3070,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_ldavg("ldavg_five")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3150,7 +3150,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_ldavg("ldavg_15")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3250,7 +3250,7 @@ int set_token_items(int file_number, cha
            if (i == get_column_block_device("areqsz")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3311,7 +3311,7 @@ int set_token_items(int file_number, cha
            } else if (i == get_column_block_device("util")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3356,12 +3356,12 @@ int set_token_items(int file_number, cha
                         set_block_device_lowest_time(time_value, utility, "util");
                     }
                     /* code for graph */
-                    if ((strstr(time_value, ":00:") != NULL) ||
-                        (strstr(time_value, ":10:") != NULL) ||
-                        (strstr(time_value, ":20:") != NULL) ||
-                        (strstr(time_value, ":30:") != NULL) ||
-                        (strstr(time_value, ":40:") != NULL) ||
-                        (strstr(time_value, ":50:") != NULL)) {
+                    if ((strstr(time_value, ":00:") != NULL) ||(strstr(time_value, "00分") != NULL) ||
+                        (strstr(time_value, ":10:") != NULL) ||(strstr(time_value, "10分") != NULL) ||
+                        (strstr(time_value, ":20:") != NULL) ||(strstr(time_value, "20分") != NULL) ||
+                        (strstr(time_value, ":30:") != NULL) ||(strstr(time_value, "30分") != NULL) ||
+                        (strstr(time_value, ":40:") != NULL) ||(strstr(time_value, "40分") != NULL) ||
+                        (strstr(time_value, ":50:") != NULL) ||(strstr(time_value, "50分") != NULL)) {
                         char str_tmp_echo[MAX_LINE_LENGTH - 100] = {'\0'};
                         char str_tmp_echo2[MAX_LINE_LENGTH] = {'\0'};
                         char str_tmp_echo3[MAX_LINE_LENGTH] = {'\0'};
@@ -3430,7 +3430,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_network("rxpck")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3496,7 +3496,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("txpck")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3562,7 +3562,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("rxkb")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3628,7 +3628,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("txkb")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3730,7 +3730,7 @@ int set_token_items(int file_number, cha
             if (i == get_column_network("rxerr")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3790,7 +3790,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("txerr")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3851,7 +3851,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("rxdrop")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3912,7 +3912,7 @@ int set_token_items(int file_number, cha
             } else if (i == get_column_network("txdrop")) {
                 if (SAR_OPTION == 'Z') {
                     t = strtod(token, &endp);
-                    if (strstr(time_value, "Average") != NULL)
+                    if ((strstr(time_value, "Average") != NULL) ||(strstr(time_value, "平均値") != NULL))
                         return 0;
                     set_tmp_val(t);
                     t = get_tmp_val();
@@ -3981,7 +3981,7 @@ int get_word_line(int file_number, char
     int PM = 0;
     /* for analyzing all data */
     if (SAR_OPTION == 'Z') {
-        if (strstr(*line, "PM") != NULL)
+        if ((strstr(*line, "PM") != NULL) ||(strstr(*line, "午後") != NULL)) 
             PM = 1;
         /* testing */
         char *str_tmp = "";
@@ -3991,7 +3991,6 @@ int get_word_line(int file_number, char
             /* this clause will be save for future use maybe */
             if (strstr(*line, "RESTART")) {
                 set_token_column(file_number, *line, "Linux", 1, SAR_OPTION);
-                /* for ps file */
                 linux_restart_count[file_number]++;
             }
             else
@@ -4002,7 +4001,7 @@ int get_word_line(int file_number, char
              */
             initialize_check_int();
         /* first, we are setting column number of interests */
-        } else if ((strstr(*line, "Average")) == 0 && (strstr(*line, "usr") || strstr(*line, "sys")   != NULL)) {
+        } else if (((strstr(*line, "Average") == 0) ||(strstr(*line, "平均値") == 0))  &&((strstr(*line, "usr") || strstr(*line, "sys") != NULL))) {
             if (MESSAGE_ONLY == 0)
                 append_list(&line_all_obj, str_tmp);
 
@@ -4075,7 +4074,7 @@ int get_word_line(int file_number, char
             PM = set_token_column(file_number, *line, "runq", 0, SAR_OPTION);
             /* this means, skip this clause on next loop, it's enough to set set_token_column() one time */
             CHK_Z = _LDAVG;
-        } else if ((strstr(*line, "Average")) == 0 && strstr(*line,"DEV") != NULL) {
+        } else if (((strstr(*line, "Average") == 0) ||(strstr(*line, "平均値") == 0))  &&(strstr(*line, "DEV") != NULL)) {
             if (MESSAGE_ONLY == 0)
                 append_list(&line_all_obj, str_tmp);
 
@@ -4083,7 +4082,7 @@ int get_word_line(int file_number, char
             PM = set_token_column(file_number, *line, "DEV", 0, SAR_OPTION);
             /* this means, skip this clause on next loop, it's enough to set set_token_column() one time */
             CHK_Z = _BLOCK_DEVICE;
-        } else if ((strstr(*line, "Average")) == 0 && strstr(*line,"rxpck/s") != NULL) {
+        } else if (((strstr(*line, "Average") == 0) ||(strstr(*line, "平均値") == 0)) &&(strstr(*line,"rxpck/s") != NULL)) {
             if (MESSAGE_ONLY == 0)
                 append_list(&line_all_obj, str_tmp);
 
@@ -4091,7 +4090,7 @@ int get_word_line(int file_number, char
             PM = set_token_column(file_number, *line, "rxpck", 0, SAR_OPTION);
             /* this means, skip this clause on next loop, it's enough to set set_token_column() one time */
             CHK_Z = _NETWORK;
-        } else if ((strstr(*line, "Average")) == 0 && strstr (*line,"rxerr/s") != NULL) {
+        } else if (((strstr(*line, "Average") == 0) ||(strstr(*line, "平均値") == 0)) &&(strstr(*line,"rxerr/s") != NULL)) {
             if (MESSAGE_ONLY == 0)
                 append_list(&line_all_obj, str_tmp);
 
@@ -4127,7 +4126,7 @@ int get_word_line(int file_number, char
                     CHK_CORES_N = 0;
                 }
                 CHECK_CPU_EACH++;
-            } else if (strstr(*line, "Average") != 0) {
+            } else if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) {
                 CHK_Z &= ~(1 << 0);  /* this means, skip this clause on next loop */
                 CHK_CORES_N = 0;
                 /* set_token_items(file_number, line, "cpu" , CHK_CORES_N, SAR_OPTION); */
@@ -4145,7 +4144,7 @@ int get_word_line(int file_number, char
             }
         /* setting utility arguments to meaningless value like -999 */
         } else if (CHK_Z == _TASKS) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 1);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4154,7 +4153,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "proc", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _PSWAP) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 2);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4163,7 +4162,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "pswpin/s", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _PAGING) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 3);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4172,7 +4171,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "fault", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _IO_TRANSFER_RATE) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 4);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4181,7 +4180,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "bread/s", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _MEMORY) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 5);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4190,7 +4189,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "kbmemfree", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _SWPUSED) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 6);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4199,7 +4198,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "kbswpfree", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _KERNEL_TABLE) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 7);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4208,7 +4207,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "dentunusd", -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _LDAVG) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 8);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4221,7 +4220,7 @@ int get_word_line(int file_number, char
          * setting each name to the object for future use 
          */ 
         } else if (CHK_Z == _BLOCK_DEVICE) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 9);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4234,7 +4233,7 @@ int get_word_line(int file_number, char
          * setting each name to the object for future use 
          */ 
         } else if (CHK_Z == _NETWORK) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 10);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4243,7 +4242,7 @@ int get_word_line(int file_number, char
                 set_token_items(file_number, line, "rxpck" , -999, SAR_OPTION, time_span, PM);
             }
         } else if (CHK_Z == _NETWORK_ERROR) {
-            if (strstr (*line, "Average") != 0)
+            if ((strstr(*line, "Average") != 0) ||(strstr(*line, "平均値") != 0)) 
                 CHK_Z &= ~(1 << 11);  /* this means, skip this clause on next loop */
             else {
                 if (MESSAGE_ONLY == 0)
@@ -4266,7 +4265,8 @@ int get_word_line(int file_number, char
             && strstr(*line, "rcvin/s") == 0 && strstr(*line, "cswch/s") == 0 
             && strstr(*line, "frmpg/s") == 0 && strstr(*line, "call/s") == 0 
             && strstr(*line, "scall/s") == 0 && strstr(*line, "totsck") == 0 
-            && strstr(*line, "total/s") == 0 && strstr(*line, "Average") == 0) {
+            && strstr(*line, "total/s") == 0 && strstr(*line, "Average") == 0
+            && strstr(*line, "平均") == 0) {
             /* no match */
             return 0;
         } else {
@@ -4566,7 +4566,7 @@ int get_word_line(int file_number, char
              * cpu and sets core number every time. core number is incremented, so, lately, core number could be known (including all line)  
              * stop calling this clause if reached up to MAX_CORE_NUMBERS, anyway, it will no be called very soon   
              */
-            if (strstr(*line, "Average") && !strstr(*line, "CPU") && (SHOW_AVG == 2) && (CHK_CORES_N <= MAX_CORE_NUMBERS)) {
+            if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&  !strstr(*line, "CPU") &&(SHOW_AVG == 2) &&(CHK_CORES_N <= MAX_CORE_NUMBERS)) {
                 append_list(&line_obj, str_tmp);
                 set_token_items(file_number, line, "cpu" , CHK_CORES_N, SAR_OPTION, time_span, PM); /* setting each cpu values to the object */
                 if (CHK_CORES_N == MAX_CORE_NUMBERS)
@@ -4577,7 +4577,7 @@ int get_word_line(int file_number, char
              * just because not knowing which block device will appear every time this clause is called,
              * setting each name to the object for future use 
              */ 
-            } else if (strstr(*line, "Average")  && !strstr(*line, "DEV") && (SHOW_AVG == 3)) {
+            } else if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&  !strstr(*line, "DEV") &&(SHOW_AVG == 3)) {
                 if (strstr(*line, "dev") != 0) {
                     append_list(&line_obj, str_tmp);
                     set_token_items(file_number, line, "DEV" , -999, SAR_OPTION, time_span, PM);
@@ -4586,17 +4586,17 @@ int get_word_line(int file_number, char
              * just because not knowing which network device will appear every time this clause is called,
              * setting each name to the object for future use 
              */ 
-            } else if (strstr(*line, "Average")  && !strstr(*line, "rxpck") && (SHOW_AVG == 4)) {
+            } else if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&  !strstr(*line, "rxpck") &&(SHOW_AVG == 4)) {
                 append_list(&line_obj, str_tmp);
                 set_token_items(file_number, line, "rxpck" , -999, SAR_OPTION, time_span, PM);
-            } else if (strstr(*line, "Average")  && !strstr(*line, "rxerr") && (SHOW_AVG == 5)) {
+            } else if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&  !strstr(*line, "rxerr") &&(SHOW_AVG == 5)) {
                 append_list(&line_obj, str_tmp);
                 set_token_items(file_number, line, "rxerr" , -999, SAR_OPTION, time_span, PM);
 
             /* Call set_token_items() here
              * this clause is for others which only have one line to be analyzed 
              */
-            } else if (strstr(*line, "Average")  && ((SHOW_AVG == 0)) && (strstr(*line, "CPU") == NULL)) {
+            } else if (((strstr(*line, "Average")) ||(strstr(*line, "平均値"))) &&((SHOW_AVG == 0)) &&(strstr(*line, "CPU") == NULL)) {
                 append_list(&line_obj, str_tmp);
                 /* setting utility arguments to meaningless value line -999 */
                 if (((CHK_A >> 1) & 1)) {
